client/src/common/components/Members/components/MemberDetails.jsx

```diff
@@ -358,6 +383,59 @@ class MemberDetails extends PureComponent {
     );
   };
 
+  renderLeaveOption = () => {
+    const {
+      _id: userId,
+      currentUser: { id: currentUserId },
+      onClose,
+      route
+    } = this.props;
+    const { isLeaveConfirmationVisible, pendingForLeaving } = this.state;
+
+    if (userId === currentUserId) {
+      return (
+        <div className="member-details__leave-box">
+          {isLeaveConfirmationVisible ? (
+            <div className="member-details__leave-box-confirmation">
+              <FormattedMessage id="common.member-details.leave-box-heading" />
+              <footer className="member-details__leave-box-confirmation-footer">
+                <PendingButton
+                  className="primary-button"
+                  onClick={this.handleCohortLeave}
+                  pending={pendingForLeaving}
```

Comment:
is pendingForLeaving necessary? It is used only here but the pending button doesn't wait for "pending" prop. It has own pending which is initialized inside it.